[freeimage] Usage problem. Typedef redefinition in MacOS

**Describe the bug**
There is a problem using FreeImage in MacOS, due to conflicting typedefs with Objetive-C header "objc/objc.h" . It fails both for x86_64 and arm64, just with a different type (see logs below)

**Environment**
- OS: MacOS Ventura 13
- Compiler: Xcode 15. Apple clang 15

**To Reproduce**
Steps to reproduce the behavior:
1. ./vcpkg install freeimage
2. Just include objc/objc.h and FreeImage.h headers in a C++ HelloWorld project.

**Expected behavior**
Build should succeed.

**Failure logs**
x86_64:
```
/Users/<········>/build_x86/vcpkg_installed/x64-osx-mega/include/FreeImage.h:158:17: error: typedef redefinition with different types ('int32_t' (aka 'int') vs 'signed char')
typedef int32_t BOOL;
                ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk/usr/include/objc/objc.h:81:25: note: previous definition is here
    typedef signed char BOOL; 
                        ^
```
arm64:
```
/Users/<········>/build_arm64/vcpkg_installed/arm64-osx-mega/include/FreeImage.h:158:17: error: typedef redefinition with different types ('int32_t' (aka 'int') vs 'bool')
typedef int32_t BOOL;
                ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk/usr/include/objc/objc.h:78:18: note: previous definition is here
    typedef bool BOOL;
                 ^
```

**Additional context**
Some refs online just from google:
- https://github.com/meganz/MEGAcmd/issues/364
- https://sourceforge.net/p/freeimage/bugs/215/


I've a PR prepared to fix the problem based on an overlay port I've been using for a while.

--- a/ports/freeimage/portfile.cmake
+++ b/ports/freeimage/portfile.cmake
@@ -20,6 +20,7 @@ vcpkg_from_sourceforge(
         use-functions-to-override-libtiff-warning-error-handlers.patch
         remove_auto_ptr.patch
         rawlib-build-fix.patch
+        typedef-xcode.patch
 )
 
 file(COPY "${CMAKE_CURRENT_LIST_DIR}/CMakeLists.txt"

--- a/ports/freeimage/typedef-xcode.patch
+++ b/ports/freeimage/typedef-xcode.patch
@@ -1,0 +1,14 @@
+diff --git a/Source/FreeImage.h b/Source/FreeImage.h
+index ad2574d..9588944 100644
+--- a/Source/FreeImage.h
++++ b/Source/FreeImage.h
+@@ -155,7 +155,9 @@ FI_STRUCT (FIMULTIBITMAP) { void *data; };
+ #ifndef _MSC_VER
+ // define portable types for 32-bit / 64-bit OS
+ #include <inttypes.h>
++#ifndef OBJC_BOOL_DEFINED
+ typedef int32_t BOOL;
++#endif
+ typedef uint8_t BYTE;
+ typedef uint16_t WORD;
+ typedef uint32_t DWORD;

--- a/ports/freeimage/vcpkg.json
+++ b/ports/freeimage/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "freeimage",
   "version": "3.18.0",
-  "port-version": 26,
+  "port-version": 27,
   "description": "Support library for graphics image formats",
   "homepage": "https://sourceforge.net/projects/freeimage/",
   "license": "GPL-2.0-only OR GPL-3.0-only OR FreeImage",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2830,7 +2830,7 @@
     },
     "freeimage": {
       "baseline": "3.18.0",
-      "port-version": 26
+      "port-version": 27
     },
     "freeopcua": {
       "baseline": "20190125",

--- a/versions/f-/freeimage.json
+++ b/versions/f-/freeimage.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "d1255d729b57a487ed7e6be58bb9feb8b8b3fd44",
+      "version": "3.18.0",
+      "port-version": 27
+    },
+    {
       "git-tree": "e7b2a65974c7375dc69526c2c5390f1c7932aa1b",
       "version": "3.18.0",
       "port-version": 26

